controller/linodemachine_controller.go

```diff
@@ -200,6 +202,19 @@ func (r *LinodeMachineReconciler) reconcile(
 	if !machineScope.LinodeMachine.ObjectMeta.DeletionTimestamp.IsZero() {
 		failureReason = cerrs.DeleteMachineError
 
+		linodeClusterKey := client.ObjectKey{
+			Namespace: machineScope.LinodeMachine.Namespace,
+			Name:      machineScope.Cluster.Spec.InfrastructureRef.Name,
+		}
+
+		if err = r.Client.Get(ctx, linodeClusterKey, machineScope.LinodeCluster); err != nil {
+			if err = client.IgnoreNotFound(err); err != nil {
+				logger.Error(err, "Failed to fetch Linode cluster")
+			}
+
+			return
+		}
```

Comment:
No naked returns, please.

```suggestion
		if err := r.Client.Get(ctx, linodeClusterKey, machineScope.LinodeCluster); client.IgnoreNotFound(err) != nil {
			return ctrl.Result{}, fmt.Errorf("get linodecluster %q: %w", linodeClusterKey, err)
		}
```

Reply:
updated